app/src/main/java/com/kks/nimblesurveyjetpackcompose/di/NetworkModule.kt

```diff
@@ -29,125 +31,133 @@ import javax.inject.Singleton
 private const val TIMEOUT_SECONDS = 15L
 private const val CHUCKER_MAX_CONTENT_LENGTH = 250000L
 
-@Suppress("TooManyFunctions")
 @Module
 @InstallIn(SingletonComponent::class)
-object NetworkModule {
+interface NetworkModule {
 
-    @Singleton
-    @Provides
-    fun providePreferenceManager(@ApplicationContext appContext: Context): PreferenceManager =
-        PreferenceManager(appContext)
+    @Binds
+    fun provideTokenAuthenticator(tokenAuthenticator: TokenAuthenticator): Authenticator
 
     @ServiceQualifier
-    @Provides
-    fun provideServiceRetrofit(
-        @ServiceQualifier okHttpClient: OkHttpClient,
-        moshi: Moshi
-    ): Retrofit {
-        return Retrofit.Builder().baseUrl(BuildConfig.BASE_URL)
-            .client(okHttpClient)
-            .addConverterFactory(MoshiConverterFactory.create(moshi))
-            .build()
-    }
+    @Binds
+    fun provideServiceAccessTokenInterceptor(accessTokenInterceptor: AccessTokenInterceptor): AccessTokenInterceptor
 
     @AuthQualifier
-    @Provides
-    fun provideAuthRetrofit(@AuthQualifier okHttpClient: OkHttpClient, moshi: Moshi): Retrofit {
-        return Retrofit.Builder().baseUrl(BuildConfig.BASE_URL)
-            .client(okHttpClient)
-            .addConverterFactory(MoshiConverterFactory.create(moshi))
-            .build()
-    }
+    @Binds
+    fun provideAuthAccessTokenInterceptor(accessTokenInterceptor: AccessTokenInterceptor): AccessTokenInterceptor
```

Comment:
We don't need two `Interceptor`s. We can reuse the same `AccessTokenInterceptor` instance. No?

Reply:
@AVI5HEK Unfortunately no. It is having dependency circle.